src/legacy/state/orders/middleware.tsx

```diff
@@ -58,46 +56,51 @@ export const popupMiddleware: Middleware<Record<string, unknown>, AppState> = (s
     const { summary, class: orderClass } = orderObject.order
 
     let popup: PopupPayload
-    if (isPendingOrderAction(action)) {
-      const hash = orderObject.order.orderCreationHash
-      if (hash) {
-        // EthFlow Order
-        popup = setPopupData(OrderTxTypes.TXN, {
-          summary,
-          status: 'submitted',
-          id: hash,
-          hash,
-        })
-        orderAnalytics('Posted', orderClass, 'EthFlow')
-      } else {
-        // Pending Order Popup
-        popup = setPopupData(OrderTxTypes.METATXN, { summary, status: 'submitted', id })
-        orderAnalytics('Posted', orderClass, 'Offchain')
-      }
-    } else if (isPresignOrders(action)) {
-      popup = setPopupData(OrderTxTypes.METATXN, { summary, status: 'presigned', id })
-      orderAnalytics('Posted', orderClass, 'Pre-Signed')
-    } else if (isCancelOrderAction(action)) {
-      // action is order/cancelOrder
-      // Cancelled Order Popup
-      popup = _buildCancellationPopup(orderObject.order)
-      orderAnalytics('Canceled', orderClass)
-    } else {
-      // action is order/expireOrder
-      // Expired Order Popup
-      popup = setPopupData(OrderTxTypes.METATXN, {
-        success: false,
+
+    const hash = orderObject.order.orderCreationHash
```

Comment:
Nitpick:
The logic is pretty complex (at least for me). Would be great to cover it by unit-tests

Reply:
https://github.com/cowprotocol/cowswap/pull/2552